Ml commons batch inference (#7899)
opensearch-project/documentation-website: _ml-commons-plugin/api/model-apis/index.md

# Model APIs

ML Commons supports the following model-level CRUD APIs:

- [Register Model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/)
- [Deploy Model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/deploy-model/)
- [Get Model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/get-model/)
- [Search Model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/search-model/)
- [Update Model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/update-model/)
- [Undeploy Model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/undeploy-model/)
- [Delete Model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/delete-model/)

# Predict APIs

> Predict APIs are used to invoke machine learning (ML) models. ML Commons supports the following Predict APIs:

Reviewer: Is the second instance of "Predict" intentionally capitalized?

Reviewer 2: Yes, capitalized since it's the API name.

- [Predict]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/train-predict/predict/) 

> - [Batch Predict]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/batch-predict/) (experimental)

Reviewer: "Batch Predict" is not capitalized in the title or H1 of the preceding file.

Reviewer 2: Right. Normally, we imply the operation in the H1 and left nav title but this is the actual API name so I capitalized. Alternatively, I can change everything to sentence case.

Reviewer: Fine as is

# Train API

> The ML Commons Train API lets you train machine learning (ML) algorithms synchronously and asynchronously:

Reviewer: ```suggestion
The ML Commons Train API lets you train ML algorithms synchronously and asynchronously:
```

- [Train]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/train-predict/train/)

To train tasks through the API, three inputs are required: 

> - Algorithm name: Must be one of a [FunctionName](https://github.com/opensearch-project/ml-commons/blob/1.3/common/src/main/java/org/opensearch/ml/common/parameter/FunctionName.java). This determines what algorithm the ML Engine runs. To add a new function, see [How To Add a New Function](https://github.com/opensearch-project/ml-commons/blob/main/docs/how-to-add-new-function.md).

Reviewer: ```suggestion
- Algorithm name: Must be a [FunctionName](https://github.com/opensearch-project/ml-commons/blob/1.3/common/src/main/java/org/opensearch/ml/common/parameter/FunctionName.java). This determines what algorithm the ML model runs. To add a new function, see [How To Add a New Function](https://github.com/opensearch-project/ml-commons/blob/main/docs/how-to-add-new-function.md).
```

- Model hyperparameters: Adjust these parameters to improve model accuracy.  

> - Input data: The data that trains the ML model, or applies the ML models to predictions. You can input data in two ways, query against your index or use a data frame.

Reviewer: ```suggestion
- Input data: The data that trains the ML model or applies it to predictions. You can input data in two ways: query against your index or use a data frame.
```
